asf_levies_model/notebooks/report_scenarios_summary_v2.py

```diff
@@ -0,0 +1,755 @@
+            gas_consumption_col="GaskWh",
+            electricity_consumption_col="ElectricitySingleRatekWh",
+            gas_tariff=gas_tariffs[scenario_name],
+            electricity_tariff=elec_tariffs[scenario_name],
+            unit_converter=1_000,
+            scheme_eligible_col="whd_eligible",
+        )
+        for row in range(len(ofgem_archetypes_df_whd_eligibility))
+    ]
+
+# Create unique names by modifying name attributes of consumers to reflect eligibility
+scenarios_set_2.insert(0, "baseline with whd eligibility")
+for scenario in scenarios_set_2:
+    for consumer in consumers[scenario]:
+        if consumer.scheme_eligible == True:
+            consumer.name = consumer.name + " Eligible"
+        elif consumer.scheme_eligible == False:
+            consumer.name = consumer.name + " Ineligible"
+        else:
+            raise ValueError("No eligibility specified")
+
+# %%
+# Apply £150 discount to eligible consumers for baseline scenario considering eligibility
+for consumer in consumers["baseline with whd eligibility"]:
+    if consumer.scheme_eligible == True:
+        consumer.apply_social_support_adjustment(
+            adjustment_fuel="electricity",
+            adjustment_parameter=-150,
+            adjustment_mode="flat adjustment",
+        )
+
+# Apply £300 discount to eligible consumers in double WHD scenarios
+for scenario in scenarios_set_2[1:3]:
+    for consumer in consumers[scenario]:
+        if consumer.scheme_eligible == True:
+            consumer.apply_social_support_adjustment(
+                adjustment_fuel="electricity",
+                adjustment_parameter=-300,
+                adjustment_mode="flat adjustment",
+            )
+
+# %% [markdown]
+# **Results dataframe for scenarios 1 (Baseline) to 5**
+
+# %%
+# Create a summary dataframe, in tidy format, for all consumer profiles and scenarios
+tidy_summary_1 = pd.DataFrame()
+for scenario in scenarios_set_1:
+    tidy = pd.concat([consumer.get_tidy_summary() for consumer in consumers[scenario]])
+
+    tidy["scenario"] = scenario
+
+    tidy_summary_1 = pd.concat([tidy_summary_1, tidy])
+
+# %%
+# Reshaping to recreate Martina's table for the summary chart
+tidy_summary_1["scenario"] = tidy_summary_1["scenario"].replace(
+    "baseline", "1. Baseline"
+)
+
+scenarios_summary_chart_1 = tidy_summary_1.pivot_table(
+    index=[
+        "Name",
+        "scenario",
+    ],
+    columns="Attribute",
+    values="Value",
+    aggfunc="first",
+).reset_index()
+
+# %%
+scenarios_summary_chart_1 = scenarios_summary_chart_1[
+    [
+        "Name",
+        "scenario",
+        "main_heating_fuel",
+        "electricity_bill",
+        "gas_bill",
+        "dual_fuel_bill",
+    ]
+].sort_values(by=["scenario", "Name"])
+
+# Create new column for bill change from baseline
+baseline = scenarios_summary_chart_1[
+    scenarios_summary_chart_1["scenario"] == "1. Baseline"
+].set_index("Name")["dual_fuel_bill"]
+
```

Comment:
Need to update these references to 'dual_fuel_bill' as I've changed these in the class to avoid ambuity of meaning.